### PR TITLE
Add init app method for custom app config

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

Adding `init_app` method to set additional configuration to `FastAPI` configuration to allow more control to app and expand.

Need this for sub-application to build proxying for multiple datasets: https://fastapi.tiangolo.com/advanced/sub-applications-proxy/

### CODE REVIEW

File: xpublish/rest.py [hunk 1]
```diff
@@ -172,7 +180,8 @@ def serve(self, host="0.0.0.0", port=9000, log_level="debug", **kwargs):
         -----
         This method is blocking and does not return.
         """
-
+        if self._app is None:
+            self.init_app()
```
Comment (jhamman) [inline xpublish/rest.py]: I think this is covered in the `@property` above.

Comment (lsetiawan) [reply xpublish/rest.py]: It is indeed. Thanks!

**Comment by lsetiawan** [comment]
Hmm... the black formatter is not happy with me. :disappointed:

**Comment by jhamman** [comment]
> Hmm... the black formatter is not happy with me. 😞

I pushed a commit that should hopefully fix this:

```black --line-length 100 --skip-string-normalization .```

File: xpublish/rest.py [hunk 2]
```diff
@@ -104,55 +104,59 @@ def zmetadata_json(self):
         )
         return Response(echunk, media_type="application/octet-stream")
 
-    @property
-    def app(self):
-        """ FastAPI app """
-        if self._app is None:
+    def init_app(self, **kwargs):
```
Comment (jhamman) [inline xpublish/rest.py]: Can you add a docstring for this method?

Also, can you point me to the API docs for the FastAPI constructor? I'm having trouble locating them on the web docs.

Comment (jhamman) [reply xpublish/rest.py]: Along these lines, do you think its worth adding a little blurb on how to customize the deployment to the docs? Something [here](https://xpublish.readthedocs.io/en/latest/tutorial.html)?

**Comment by lsetiawan** [comment]
FastAPI Configuration: https://fastapi.tiangolo.com/tutorial/application-configuration/
Sub Application in FastAPI: https://fastapi.tiangolo.com/advanced/sub-applications-proxy/

**Comment by jhamman** [comment]
Thanks @lsetiawan!

**Pull request merged**